File: 9.palindromeNumber/palindromeNumber.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        # strX = str(x)
        # reversedX = strX[::-1]

        # return True if strX == reversedX else False

        if x < -2**21 or 2**31 - 1 < x:
            return False
        
        if x < 0 or (x % 10 == 0 and x != 0):
            return False
        
        reversedHalf = 0

        while x > reversedHalf:
            reversedHalf = reversedHalf * 10 + x % 10
            x //= 10

        return x == reversedHalf or x == reversedHalf // 10

File: 9.palindromeNumber/palindromeNumber.php
<?
class Solution {
    function isPalindrome($x) {
        // $strX = strval($x);
        // $revX = strrev($strX);

        // return $strX == $revX ? true : false;

        if ($x < -2**31 || 2**31 -1 < $x) return false;

        if ($x < 0 || ($x % 10 == 0 && $x != 0)) return false;

        $reverseHalf = 0;

        while ($x > $reverseHalf) {
            $reverseHalf = $reverseHalf * 10 + $x % 10;
            $x = intdiv($x, 10);
        }

        return $x == $reverseHalf || $x == intdiv($reverseHalf, 10);
    }
}